### PR TITLE
fix a bug causing 'spawn.before' to be truncated w/ searchwindowsize use

### PR DESCRIPTION
**Pull request by ryanpetrello**
see: https://github.com/pexpect/pexpect/issues/478

### CODE REVIEW

File: pexpect/expect.py [hunk 1]
```diff
@@ -27,11 +28,11 @@ def new_data(self, data):
             window = spawn.buffer
         index = searcher.search(window, len(data))
         if index >= 0:
-            value = spawn.buffer
             spawn._buffer = spawn.buffer_type()
-            spawn._buffer.write(value[searcher.end:])
-            spawn.before = value[: searcher.start]
-            spawn.after = value[searcher.start: searcher.end]
+            spawn._buffer.write(window[searcher.end:])
```
Comment (ryanpetrello) [inline pexpect/expect.py]: @takluyver these search match indexes are _actually_ relative to `window`, which _may not_ be equal to `spawn.buffer` when `searchwindowsize` is in use.

**Comment by takluyver** [comment]
Thanks for the investigation and speedy fix. I think what you're doing here makes sense.

I don't quite know what you mean about other bugs - do you think they're fixed by this PR? Is there something more you still want to do? I'm happy to merge this if you're happy with it.

**Comment by ryanpetrello** [comment]
> I don't quite know what you mean about other bugs

This PR _also_ changes how I set `spawn.after` to use `window` instead of `value`; I think without this, people will _also_ see bugs on that property similar to what https://github.com/pexpect/pexpect/issues/478 describes for `before`.

**Comment by takluyver** [comment]
Aha, gotcha. Merging now, thanks!

**Pull request merged**